google/cloud/spanner/client.h

```diff
@@ -374,13 +374,16 @@ class Client {
    * Can also execute a DML statement with a returning clause in a read/write
    * transaction.
    *
-   * @par Example with explicitly selected columns.
+   * ### Example with explicitly selected columns.
+   *
    * @snippet samples.cc spanner-query-data
    *
-   * @par Example using `SELECT *`.
+   * ## Example using `SELECT *`.
```

Comment:
Why `##` when the others are `###`?

Reply:
Because I neglected to make another pass after finding workarounds for `@par` and the backticks.

I have now kept all of these as `@par`, but I moved the text with backticks to the body line.  I also fixed the paragraphs.  A `@par` command stops on the first blank line, which stops the indentation, note how these examples are indented:

https://googleapis.dev/cpp/google-cloud-spanner/latest/classgoogle_1_1cloud_1_1spanner_1_1Client.html#a8e2afee42f535c0436d9161c54b84179

vs. these:

https://googleapis.dev/cpp/google-cloud-spanner/latest/classgoogle_1_1cloud_1_1spanner_1_1Value.html#details

I also found some inconsistencies in the location of the examples vs. the notes and warnings vs. the parameters.

They now (should) go: (1) the notes and warnings, (2) the examples, (3) the parameters, return types, and anything about `@throw`. That works well with `@copydoc`, and I think logically it is a better order too.